### PR TITLE
FastMNMF was refactored and FastMNMF2 was added

### PR DESCRIPTION
**Pull request by sekiguchi92**
Thanks for sending a pull request (PR), we really appreciate that! Before hitting the submit button, we'd be really glad if you could make sure all the following points have been cleared.

Please also refer to the doc on [contributing](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html) for more details. Even if you can't check all the boxes below, do not hesitate to go ahead with the PR and ask for help there.

- [x] Are there docstrings ? Do they follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) style ?
- [x] Have you run the tests by doing `nosetests` or `py.test` at the root of the repo ?
- [x] Have you checked that the doc builds properly and that any new file has been added to the repo ? How to do that is covered in the [documentation](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html#documentation).
- [x] Is there a unit test for the proposed code modification ? If the PR addresses an issue, the test should make sure the issue is fixed.
- [x] Last but not least, did you document the proposed change in the CHANGELOG file ? It should go under "Unreleased".

Happy PR :smiley:

### CODE REVIEW

**Comment by fakufaku** [comment]
@sekiguchi92 Thank you so much for the contribution! Sorry it took so long for me to reply here!

There is only a small thing before I merge. We have moved to enforcing formatting by black on the whole codebase. Would you mind doing ?

**Comment by sekiguchi92** [comment]
Thank you for your reply.

I had already used black before PR.
And I tried to apply black again just now, but did not get any change.

**Comment by fakufaku** [comment]
Hum, the action says that files `pyroomacoustics/bss/fastmnmf.py` and `pyroomacoustics/bss/fastmnmf2.py` are not formatted correctly. The version of black is `21.12b0`. Could you give it another try ? If it doesn't work this time I can take care of it.
```
  Verifying setup for Black…
  Verified Black setup
  Will use Black to check the files with extensions py
  Linting files in /home/runner/work/pyroomacoustics/pyroomacoustics with Black…
  Skipping .ipynb files as Jupyter dependencies are not installed.
  You can fix this by running ``pip install black[jupyter]``
  would reformat pyroomacoustics/bss/fastmnmf.py
  would reformat pyroomacoustics/bss/fastmnmf2.py
  Oh no! 💥 💔 💥
  2 files would be reformatted, 149 files would be left unchanged.
  Black found 2 errors (failure)
```

**Comment by sekiguchi92** [comment]
I'm sorry I found that black version was old (19.1). After updating black (22.1), fastmnmf.py and fastmnmf2.py were reformatted.

I'm sorry for the trouble.

File: pyroomacoustics/bss/fastmnmf2.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
+# Copyright (c) 2019 Kouhei Sekiguchi, Yoshiaki Bando
```
Comment (fakufaku) [inline pyroomacoustics/bss/fastmnmf2.py:1]: Small detail, but shouldn't this be 2021 or 2022 ?

**Review by fakufaku — APPROVED** [review]
LGTM. Let me know if you want to update the copyright date.

**Pull request merged**